### PR TITLE
feat: flip images / eq method for image

### LINKED ISSUE #280: Flip image horizontally and vertically
### Is your feature request related to a problem?

A common transformation on images is to flip them either horizontally or vertically.

### Desired solution

Add two methods to `Image`:
* `flip_horizontally` to flip the image horizontally
* `flip_vertically` to flip the image vertically

Both should return a new image and leave the original unchanged.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

image

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by patrikguempel**
flip images

Closes #280.

###Summary of Changes
Images can now be flipped both horizontally and vertically.
Image now has an eq method 

Co-authored-by: daniaHu <129186516+daniaHu@users.noreply.github.com>

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/7.0.4) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/5288022440)

| Descriptor  |                                Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|-----------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/7.0.4/descriptors/python_black)          |    2|    2|                                                             0|0.77s       |
|❌ PYTHON    |[mypy](https://megalinter.io/7.0.4/descriptors/python_mypy)            |    2|     |[7](https://github.com/Safe-DS/Stdlib/actions/runs/5288022440)|1.77s       |
|❌ PYTHON    |[ruff](https://megalinter.io/7.0.4/descriptors/python_ruff)            |    2|    2|[3](https://github.com/Safe-DS/Stdlib/actions/runs/5288022440)|0.02s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/7.0.4/descriptors/repository_git_diff)|yes  |     |no                                                            |0.01s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5288022440)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/image/containers/_image.py [hunk 1]
```diff
@@ -158,7 +178,7 @@ def _repr_png_(self) -> bytes | None:
 
     def resize(self, new_width: int, new_height: int) -> Image:
         """
-        Return the resized image.
+        Resize an image to any size.
```
Comment (alex-senger) [inline src/safeds/data/image/containers/_image.py:181]:
```suggestion
        Return an image that has been resized to a given size.
```

File: tests/safeds/data/image/containers/test_image.py [hunk 2]
```diff
@@ -223,15 +223,27 @@ def test_should_raise(self) -> None:
...
+
+class TestFlipHorizontally:
     def test_should_flip_horizontally(self) -> None:
         image = Image.from_png_file(resolve_resource_path("image/original.png"))
-        image = image.flip_image_horizontally()
+        image = image.flip_horizontally()
         image2 = Image.from_png_file(resolve_resource_path("image/flip_horizontally.png"))
         assert image == image2
+
+    def test_should_should_be_original(self) -> None:
```
Comment (Marsmaennchen221) [inline tests/safeds/data/image/containers/test_image.py:246]:
```suggestion
    def test_should_be_original(self) -> None:
```

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

**Review by Marsmaennchen221 — APPROVED** [review]

**Pull request merged**